### PR TITLE
{Service Connector} Add input validation for `run_cli_cmd` 

### PR DESCRIPTION
**Pull request by xfz11**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Connector

**Review by yungezz — APPROVED** [review]

**Comment by jsntcy** [comment]
Please our centralized function `run_cmd` with security consideration instead of writing your own.
More details in https://github.com/Azure/azure-cli/blob/282c6e8f4358934094b512e29ae1421438de67aa/doc/cli_subprocess_guidelines.md#cli-centralized-subsystem-executing

@AllyW

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_transformers.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def transform_linker_properties(result):
     if is_aks_linker_by_id(resource_id):
         result['kubernetesResourceName'] = get_aks_resource_name(result)
     try:
-        output = run_cli_cmd('az webapp connection list-configuration --id {} -o json'.format(resource_id))
+        output = run_cli_cmd('az webapp connection list-configuration --id "{}" -o json'.format(resource_id))
         result['configurations'] = output.get('configurations')
```
Comment (AllyW) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_transformers.py:61]: Hi @xfz11 , considering potential security risks subprocess might bring, cli recently provided a centralized func `run_cmd` to deal with these issues. 

code is here: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/util.py#L1361-L1372

More guidelines about subprocess security vulnerabilities and fix examples can be found in this doc https://github.com/Azure/azure-cli/blob/282c6e8f4358934094b512e29ae1421438de67aa/doc/cli_subprocess_guidelines.md#cli-centralized-subsystem-executing

Please follow the example and fix related issues.

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_transformers.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def transform_local_linker_properties(result):
     result = todict(result)
     resource_id = result.get('id')
     try:
-        output = run_cli_cmd('az connection generate-configuration --id {} -o json'.format(resource_id))
+        output = run_cli_cmd('az connection generate-configuration --id "{}" -o json'.format(resource_id))
         result['configurations'] = output.get('configurations')
```
Comment (AllyW) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_transformers.py:81]: Same as above.

Please following this link https://github.com/Azure/azure-cli/blob/282c6e8f4358934094b512e29ae1421438de67aa/doc/cli_subprocess_guidelines.md#cli-centralized-subsystem-executing to fix these issues.

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py [hunk 3]
```diff
@@ -148,7 +148,7 @@ def _infer_webapp(source_id):
 
         client_type = None
         try:
-            output = run_cli_cmd('az webapp show --id {} -o json'.format(source_id))
+            output = run_cli_cmd('az webapp show --id "{}" -o json'.format(source_id))
             prop = output.get('siteConfig').get('linuxFxVersion', None) or\
```
Comment (AllyW) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py:152]: Same as above.

**Comment by AllyW** [comment]
> @AllyW Could you help review and merge the PR?

For input validation from business logic, please involve related team members to review and confirm it first.

**Review by brycechen1849 — APPROVED** [review]

**Comment by AllyW** [comment]
Proper input validation is necessary for reducing command risks, but for `az` related cmds in cli, internal cli function invoking is recommended to avoid script injections from `shell=True` of `subprocess`. For more info, please check doc here: https://github.com/Azure/azure-cli/blob/5d82c39f56803de73eff57fc6c337b07c854f521/doc/cli_subprocess_guidelines.md#notes

**Review by AllyW — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**